src/SearchField/SearchFieldClearButton.jsx

```diff
@@ -0,0 +1,22 @@
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+
+import { SearchFieldContext } from './SearchFieldAdvanced';
+
+const SearchFieldClearButton = () => {
+  const {
+    screenReaderText, icons, value,
+  } = useContext(SearchFieldContext);
+
+  return (
+    <button
+      type="reset"
+      className={classNames('btn px-2', { 'd-none': !value })}
```

Comment:
Random thought, feel free to ignore. But you could also early return `null` if there's no value instead of toggling a class and leaving the button in the DOM.

Reply:
True. The previous implementation of this component removes the button from the DOM which caused styling issues. However, this time around, there are no styling issues if the button is removed the DOM altogether.

I will early return `null` here instead.